runtime-light/server/http/init-functions.cpp

```diff
@@ -0,0 +1,188 @@
+constexpr auto *HEAD_METHOD = "HEAD";
+
+constexpr auto *ENCODING_GZIP = "gzip";
+constexpr auto *ENCODING_DEFLATE = "deflate";
+
+[[maybe_unused]] constexpr auto *CONTENT_TYPE = "CONTENT_TYPE";
+
+string get_server_protocol([[maybe_unused]] tl::HttpVersion http_version, [[maybe_unused]] const std::optional<string> &opt_scheme) noexcept {
+  return string{"HTTP/1.1"}; // TODO
+}
+
+void process_cookie_header(const string &header, PhpScriptBuiltInSuperGlobals &superglobals) noexcept {
+  // *** be careful here ***
+  auto *cookie_start{const_cast<char *>(header.c_str())};
+  auto *cookie_list_end{const_cast<char *>(header.c_str() + header.size())};
+  do {
+    auto *cookie_end{std::find(cookie_start, cookie_list_end, ';')};
+    char *cookie_domain_end{std::find(cookie_start, cookie_end, '=')};
+    if (cookie_domain_end != cookie_end) [[likely]] {
+      char *cookie_value_start{std::next(cookie_domain_end)};
+      const auto cookie_domain_size{static_cast<string::size_type>(std::distance(cookie_start, cookie_domain_end))};
+      const auto cookie_value_size{static_cast<string::size_type>(std::distance(cookie_value_start, cookie_end))};
+      string cookie_domain{cookie_start, cookie_domain_size};
+      string cookie_value{cookie_value_start, cookie_value_size};
+      superglobals.v$_COOKIE.set_value(cookie_domain, std::move(cookie_value)); // TODO use proper setter
+    }
+    // skip ';' if possible
+    cookie_start = cookie_end == cookie_list_end ? cookie_list_end : std::next(cookie_end);
+    // skip whitespaces
+    while (cookie_start != cookie_list_end && *cookie_start == ' ') {
+      cookie_start = std::next(cookie_start);
+    }
+  } while (cookie_start != cookie_list_end);
+}
+
+void process_headers(tl::dictionary<tl::httpHeaderValue> &&headers, PhpScriptBuiltInSuperGlobals &superglobals) noexcept {
+  auto &server{superglobals.v$_SERVER};
+  auto &http_server_ctx{HttpServerComponentContext::get()};
+  using namespace PhpServerSuperGlobalIndices;
+
+  // platform provides headers that are already in lowercase
+  for (auto &[header_name, header] : headers) {
+    if (std::strcmp(header_name.c_str(), HEADER_ACCEPT_ENCODING) == 0) {
+      if (std::strstr(header.value.c_str(), ENCODING_GZIP) != nullptr) {
+        http_server_ctx.encoding |= HttpServerComponentContext::ENCODING_GZIP;
+      }
+      if (std::strstr(header.value.c_str(), ENCODING_DEFLATE) != nullptr) {
+        http_server_ctx.encoding |= HttpServerComponentContext::ENCODING_DEFLATE;
+      }
+    } else if (std::strcmp(header_name.c_str(), HEADER_COOKIE) == 0) {
+      process_cookie_header(header.value, superglobals);
+    } else if (std::strcmp(header_name.c_str(), HEADER_HOST) == 0) {
+      server.set_value(string{SERVER_NAME, std::char_traits<char>::length(SERVER_NAME)}, header.value);
+    } else if (std::strcmp(header_name.c_str(), HEADER_AUTHORIZATION) == 0) {
+      // TODO parse authorization
+    }
+    // add header entries
```

Comment:
In [`runtime/interface.cpp`](https://github.com/VKCOM/kphp/blob/master/runtime/interface.cpp#L1648) this block stored in else branch

Reply:
The semantics is equal in fact